### ISSUE TITLE
[Intermittent] The “New Tab Added!” and “Saved to Bookmarks!” notifications are displayed above where the menu tray would be when the keyboard is displayed

### ISSUE THREAD

**Issue by heres-maria** [MozillaReality/FirefoxReality#2375]
**[Affected Versions]:**
- Firefox Reality version 7 RC3 (7b66fb14)

**[Affected Platforms]:**
- HTC Vive Focus Plus - Android 7.1.2 
- Oculus Quest - Android 7.1.1

**[Prerequisites]:**
- The headset is turned on and the browser is open.

**[Steps to reproduce]:**
1. Go to a webpage that contains an input field (e.g. [YouTube](https://www.youtube.com/)).
2. Click inside the input field so that the keyboard appears.
3. While the keyboard is displayed bookmark the page.
4. Observe the position of the notification.

**[Expected results]:**
- When displaying notifications, the UI takes into account that the menu tray is not displayed at the same time as the keyboard.

**[Actual results]:**
- The notification is displayed above where the menu tray would be.

**[Notes]:**
- This issue is reproducible for the “New Tab Added!” notification as well.
- Attached is a recording of the [issue](https://drive.google.com/file/d/1GQImZbQQFcn_Nfx4Ed_jcF5NLd-5kcSD/view?usp=sharing).

**Comment by keianhzo** [comment]
@jvonitter not sure if we ever considered that case. I'd say that hiding the keyboard is the best option?

**Comment by jvonitter** [comment]
@thenadj what do you think?

**Comment by jvonitter** [comment]
same solution as in #2564 -
let's move the notification to be centered below the transport controls when the tray is not visible. Also remove the pointy part that sticks out of the bubble (so it doesn't point anywhere).

**Comment by keianhzo** [comment]
@jvonitter For this specific issue, if we show the notification under the navigation bar instead of the tray, it is still occluded by the keyboard (in case it is in the default position) as they keyboard remains visible when you bookmark the page if an input field is focused.

One alternative would be to always hide the keyboard when showing notifications so they are never occluded.

**Comment by jvonitter** [comment]
@keianhzo i dont think we should hide the keyboard. if it's there, it's because the user took some action for it to appear. what do you think about displaying it in the center of the window? could do the same thing with #2564 , so it's consistent

**Comment by keianhzo** [comment]
@jvonitter The main issue I see with that is that the keyboard could be exactly at the same position as the notification and occlude it. Actually anywhere we place it, the keyboard could be on top if we don't hide it. I'll share an APK with the above implementation so you can see it we can discuss further.

**Comment by heres-maria** [comment]
I have verified this issue and it is no longer reproducible with the latest Firefox Reality version 9-40371914 (083b65c0) build.
The notifications are now displayed in the center of the active window when the tray is not displayed. None of them have the little triangle anymore. 
Tested on Oculus Go (Android 7.1.2), Oculus Quest (Android 7.1.1), HTC Vive Focus Plus (Android 7.1.2) and HTC Vive Focus (Android 7.1.2).
